.github/workflows/ci.yaml

```diff
@@ -0,0 +1,25 @@
+name: Java CI
+
+on: [push, pull_request]
```

Comment:
Recommend adding `workflow_dispatch` to allow manual triggering, and limiting pushes to `master` branch.
```
on: 
  workflow_dispatch:
  pull_request:
  push:
    branches:
      - master
```

Reply:
If i get that one properly, it would not normally auto run on other branches (say on a fork with many as I just did to raise a bunch of PRs - but would on pull request for sure) unless going into the workflow_dispatch and triggering said branch.  I had not seen that one but looked on oshi and sort of understand that and just wanted to clarify my assumptions.  Assuming I have that correct, sounds like a good idea as most contributors will never have anything but master anyways ;)

Reply:
Yep. 

As I wrote it above the branch restriction only applies to "push" to the repository, so doesn't stop any PR runs.  

I think it's always a good idea to add `workflow_dispatch`.

The master branch restriction may not be relevant here, but there is a `generate-site` branch that can be pushed to, and there's no need to run actions when that happens.  Adding `workflow_dispatch` allows you to manually choose to run it on any other branches if needed.

Reply:
> Recommend adding `workflow_dispatch` to allow manual triggering, and limiting pushes to `master` branch.
👍

Reply:
> The master branch restriction may not be relevant here, but there is a `generate-site` branch that can be pushed to

This branch cannot be used: this is a temporary branch that is automatically generated by a GA after a PR is merged in order to generate the snapshot doc.

Any change to the master branch will be detected by Gihub and it will re-generate the Jekyll website.

Reply:
> This branch cannot be used: this is a temporary branch that is automatically generated by a GA after a PR is merged

I know this PR has been merged, but I did want to point out that any branch, temporary or otherwise, generated by anyone (human or automation) qualifies as a "push" to the repository and would trigger a build.  It's not a big deal if it's only done rarely (and it's free anyway).

Speaking of free and to close the loop on billing, GHA is free with no quota for public repos like this one.  I migrated my project over two years ago (well, @hazendaz did!) and [have run 3400 workflows since then](https://github.com/oshi/oshi/actions), most in the 5 minute range, some taking hours.  I haven't paid a dime.  GHA is owned by Microsoft and backed by the massive Azure cloud, so resource limits and backlogs aren't really a thing.

Travis claims to be free but allocates minutes to OSS projects that must be applied for, but more importantly they have a limited number of machines and I honestly don't know how long they'll be around.  I still use Travis only for an ARM hardware build because GHA doesn't have those (yet... guess what one of the things I'm supporting at work is) and ARM provides those machines for free.

Reply:
@dbwiddis : I've just merged this PR and did the updates in a subsequent commit after ;-) You can have a look at the code in master now. It's clean IMO. Let me know 👍